app/controllers/activities_controller.rb

```diff
@@ -1,6 +1,8 @@
 class ActivitiesController < ApplicationController
+  before_action :authenticate_user!
+
  def index
- 	sleep 0.7
+    sleep 0.7
     @activities = PublicActivity::Activity.order("created_at desc").where(recipient_id: current_user).where("owner_id not in (?)", current_user).all
   end
 end
```

Comment:
@sonalkr132 I can't find any way to define ability for index action of activities.Searched a lot but couldn't find any related answers.
I put this line in ability.rb in `permitted(user)` method  `can [:index], PublicActivity::Activity` 
and used `authorize! :index, @activities` for authorization in activities_controller.
Can you point me where i am going wrong?Its not giving any error.guest user can't access activities.But logged in user also can't access activities.